app/models/module_item.rb

```diff
@@ -26,14 +17,82 @@ def self.load_file
     data
   end
 
-  # Returns all the module items that belong to a particular topic within a training module
-  scope :where_topic, lambda { |training_module, topic|
-    pattern = /\A(\d+\W){2}#{topic}(?=(\D|$))/ # Start with two number then non-word character pairs (e.g. 2-4- or 13.4.). Can be followed by either a non-digit or end of line
+  # @return [Hash] 'Type' to 'View object' mapping
+  MODELS = {
+    module_intro: ContentPage,
+    sub_module_intro: ContentPage,
+    text_page: ContentPage,
+    youtube_page: YoutubePage,
+    formative_assessment: Questionnaire,
+  }.freeze
+
+  # @return [Regexp] 2nd digit if present: 1-[1]-1-1
+  SUBMODULE_PATTERN = %r"\A(?<prefix>\d+\W){1}(?<submodule>\d+)(?=(?<suffix>\D|$))"
+
+  # @return [Regexp] 3rd digit if present: 1-1-[1]-1
+  TOPIC_PATTERN = %r"\A(?<prefix>\d+\W){2}(?<topic>\d+)(?=(?<suffix>\D|$))"
+
+  # @return [Regexp] 4th digit (and optional suffix) if present: 1-1-1-[1a]
+  # PAGE_PATTERN = %r"\A(?<prefix>\d+\W){3}(?<page>\d+)(?=(?<suffix>\D|$))".freeze
+  PAGE_PATTERN = %r"\A(?<prefix>\d+\W){3}(?<page>\d+\D+)$"
+
```

Comment:
someday you must share your REGEX spells

Reply:
Haha, wasn't perfect, last `+` needed to be a `*`. Tada!